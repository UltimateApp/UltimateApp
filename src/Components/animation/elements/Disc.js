import React from 'react';
import { StyleSheet, View, Text } from 'react-native';

import theme from '../../../styles/theme.style';

const Disc = ({ width, number }) => {
  const itemStyle = [
    styles.disc,
    {
      height: width,
      width,
      borderRadius: width,
      borderWidth: width / 10,
    },
  ];

  return (
    <View style={itemStyle}>
      <Text style={styles.discText}>{number}</Text>
    </View>
  );
};

export default Disc;

const styles = StyleSheet.create({
  disc: {
    marginRight: 10,
    textAlign: 'center',
    textAlignVertical: 'center',
    alignItems: 'center',
    justifyContent: 'center',
    borderColor: theme.DISC_BORDER,
    backgroundColor: theme.DISC_COLOR,
  },
  discText: {
    fontWeight: 'bold',
    color: theme.DISC_COLOR, //Make the number in the disc invisible,
  },
});
