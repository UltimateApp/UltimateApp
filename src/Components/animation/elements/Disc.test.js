import React from 'react';
import { create, act } from 'react-test-renderer';

import Disc from './Disc';

describe('<Disc />', () => {
  it('renders correctly', async () => {
    const tree = create(<Disc baseWidth={10} number={1} />).toJSON();
    await act(async () => expect(tree).toMatchSnapshot());
  });
});
