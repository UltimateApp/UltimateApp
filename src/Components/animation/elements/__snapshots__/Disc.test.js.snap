// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Disc /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "black",
        "justifyContent": "center",
        "marginRight": 10,
        "textAlign": "center",
        "textAlignVertical": "center",
      },
      Object {
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 10,
        "width": 10,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontWeight": "bold",
      }
    }
  >
    1
  </Text>
</View>
`;
